components/apim-analytics/org.wso2.analytics.solutions.apim.analytics/widgets/APIMOverallApiStats/src/APIMOverallApiStatsWidget.jsx

```diff
@@ -0,0 +1,342 @@
+        super.getWidgetChannelManager().subscribeWidget(this.props.id, this.handleTopAPIReceived, dataProviderConfigs);
+    }
+
+    /**
+     * Formats data received from assembleTopAPIQuery
+     * @param {object} message - data retrieved
+     * @memberof APIMOverallApiStatsWidget
+     * */
+    handleTopAPIReceived(message) {
+        const { data } = message;
+
+        if (data) {
+            this.setState({ topApiIdData: data });
+        }
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
+        this.assembleAPIDataQuery();
+    }
+
+    /**
+     * Formats the siddhi query - apilistquery
+     * @memberof APIMOverallApiStatsWidget
+     * */
+    assembleAPIDataQuery() {
+        const { providerConfig } = this.state;
+
+        const dataProviderConfigs = cloneDeep(providerConfig);
+        dataProviderConfigs.configs.config.queryData.query = dataProviderConfigs.configs.config.queryData.apilistquery;
+        super.getWidgetChannelManager().subscribeWidget(this.props.id, this.handleAPIDataReceived, dataProviderConfigs);
+    }
+
+    /**
+     * Formats data received from assembleAPIDataQuery
+     * @param {object} message - data retrieved
+     * @memberof APIMOverallApiStatsWidget
+     * */
+    handleAPIDataReceived(message) {
+        const { data } = message;
+        const { topApiIdData } = this.state;
+        if (data) {
+            let counter = 0;
+            const topApiNameData = [];
+            topApiIdData.forEach((apiIdData) => {
+                counter += 1;
+                const apiID = apiIdData[0];
+                let apiName = '';
+                data.forEach((dataUnit) => {
+                    if (dataUnit[0] === apiID) {
+                        apiName = dataUnit[1] + ' ' + dataUnit[2];
+                    }
+                });
+                topApiNameData.push({ id: counter, apiname: apiName, ratings: apiIdData[1] });
+            });
+            this.setState({ topApiNameData });
+        }
+        super.getWidgetChannelManager().unsubscribeWidget(this.props.id);
+    }
+
+    /**
+     * @inheritDoc
+     * @returns {ReactElement} Render the APIM Overall Api Stats widget
+     * @memberof APIMOverallApiStatsWidget
+     */
+    render() {
+        const {
+            localeMessages, faultyProviderConfig, height, availableApiData, legendData, topApiNameData,
+        } = this.state;
+        const {
+            loadingIcon, paper, paperWrapper, headingWrapper,
+        } = this.styles;
+        const themeName = this.props.muiTheme.name;
+        const availabilityProps = { availableApiData, legendData };
+
+        if (!localeMessages) {
+            return (<CircularProgress style={loadingIcon} />);
+        }
+        return (
+            <IntlProvider locale={languageWithoutRegionCode} messages={localeMessages}>
+                {
+                    faultyProviderConfig ? (
+                        <div
```

Comment:
Why this block is not inside a `<MuiThemeProvider>`? Since you have used material-ui components here, you should warp them with `<MuiThemeProvider>`.

Reply:
Also, in order to improve readability lets we move these two blocks in the ternary operator into 2 `render` functions.

Reply:
Wrapped the content pointed in the comment using <MuiThemeProvider>

Reply:
I have improved the readability by breaking the bulk code into two separate components.